OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,1142 @@
+    col_ind.erase(old_collabel);
+    col_ind[new_collabel] = colind;
+  }
+
+  /// Update all the column labels. Returns an iterator pair(std::pair) where
+  /// first element of pair is the beginning and second element of the pair
+  /// is the end(sentinel) of the labels. Dereferencing the iterator will 
+  /// produce a pair(std::pair) where the first element of the pair is the 
+  /// label and the second element is the column index.
+  ColLabelsIterPairType updColLabels() {
+    return std::make_pair(col_ind.begin(), col_ind.end());
+  }
+
+  /// Get the index of a column from its label. Time complexity is constant on
+  /// average and linear in number of column labels on worst case.
+  ///
+  /// \throws OpenSim::ColumnDoesNotExist If the column label does not exist.
+  size_t getColInd(const string& collabel) const {
+    try {
+      return col_ind.at(collabel);
+    } catch(const std::out_of_range&) {
+      throw ColumnDoesNotExist{"No column with label '" + collabel + "'."};
+    }
+  }
+
+  /// Clear all the column labels.
+  void clearColLabels() {
+    col_ind.clear();
+  }
+
+private:
+  // Helper function. Check if a column exists and throw an exception if it
+  // does not.
+  void checkColExists(const size_t colind) const {
+    if(!colExists(colind)) {
+      throw ColumnDoesNotExist{"Column " + std::to_string(colind) + 
+                               " does not exist."};
+    }
+  }
+
+  // Helper function. Check if a column has label and throw an exception if it
+  // does not.
+  void checkColHasLabel(const size_t colind) const {
+    if(colHasLabel(colind)) {
+      throw ColumnHasLabel{"Column " + std::to_string(colind) + 
+                           " already has a label."};
+    }
+  }
+
+  // Helper function. Does both checkColExists() and checkColHasLabel().
+  void checkColExistsAndHasLabel(const size_t colind) const {
+    checkColExists(colind);
+    checkColHasLabel(colind);
+  }
+
+  // Helper function. Round to next highest power of 2. Works only for 32 bits.
+  size_t rndToNextPowOf2(size_t num) {
+    --num;
+    num |= (num >>  1); // Highest  2 bits are set by end of this.
+    num |= (num >>  2); // Highest  4 bits are set by end of this.
+    num |= (num >>  4); // Highest  8 bits are set by end of this.
+    num |= (num >>  8); // Highest 16 bits are set by end of this.
+    num |= (num >> 16); // Highest 32 bits are set by end of this.
+    return ++num;
+  }
+
+  /// Matrix of data. This excludes timestamp column.
+  SimTK::Matrix_<ET> data;
+  /// Meta-data.
+  MetaDataType       metadata;
+  /// Column label to column index.
+  ColLabelsType      col_ind;
+};  // DataTable_
+
+
+/// Bind two DataTable_(s) by row and produce a new DataTable_.
+template<typename ET>
+OpenSim::DataTable_<ET> 
+OpenSim::rbindDataTables(const OpenSim::DataTable_<ET>& dt1, 
+                         const OpenSim::DataTable_<ET>& dt2) {
+  OpenSim::DataTable_<ET> dt{dt1};
+  dt.rbindDataTable(dt2);
+  return dt;
```

Comment:
Are all compilers smart enough or should you write `return std::move(dt);` here?